timemachine/cpp/src/bound_potential.cu

```diff
@@ -37,14 +37,30 @@ void BoundPotential::execute_host(
     gpuErrchk(cudaMemset(d_du_dl.data, 0, d_du_dl.size));
     gpuErrchk(cudaMemset(d_u.data, 0, d_u.size));
 
-    this->execute_device(
-        N, d_x.data, d_box.data, lambda, d_du_dx.data, nullptr, d_du_dl.data, d_u.data, static_cast<cudaStream_t>(0));
-
+    cudaStream_t stream = static_cast<cudaStream_t>(0);
+    this->execute_device(N, d_x.data, d_box.data, lambda, d_du_dx.data, nullptr, d_du_dl.data, d_u.data, stream);
+    gpuErrchk(cudaStreamSynchronize(stream));
     d_du_dx.copy_to(h_du_dx);
     d_du_dl.copy_to(h_du_dl);
     d_u.copy_to(h_u);
 };
 
+void BoundPotential::set_params_device(
+    const std::vector<int> device_shape, const double *d_new_params, const cudaStream_t stream) {
```

Comment:
nit: could this take a `DeviceBuffer<double>` instead of the raw buffer?
```suggestion
    const std::vector<int> device_shape, const DeviceBuffer<double> &new_params, const cudaStream_t stream) {
```

Reply:
Would `new_shape` be a more descriptive name than `device_shape`?

Reply:
Right now our convention is that the `_device` methods take raw pointers. Not sure I would want to change it anyways in the case that you have say a slice of a DeviceBuffer.